compiler/modules/print.llvm/llvm.bal

```diff
@@ -287,47 +297,124 @@ public class FunctionDefn {
         return tem;
     }
 
-    function genLabel() returns string {
-        string label = "L" + self.labelCount.toString();
+    function genLabel() returns int {
+        int label = -(self.labelCount + 1);
         self.labelCount += 1;
         return label;
     }
 
-    function genReg(string? name = ()) returns string {
-        string regName = "_" + self.unnamedRegisterCount.toString();
+    function genReg(string? name = ()) returns string|int {
         if name is string {
-            regName = name;
+            string regName = name;
+            if self.variableNames.hasKey(regName) {
+                int count = self.variableNames.get(regName);
+                self.variableNames[regName] = count + 1; // increment the count of the base name
+                string newName = regName + "." + count.toString();
+                while self.variableNames.hasKey(newName) {
+                    count += 1;
+                    newName = regName + "." + count.toString();
+                }
+                regName = newName;
+                self.variableNames[regName] = 1; // save the augmented name in case user use the same  
+            }
+            else {
+                self.variableNames[regName] = 1;
+            }
+            regName = self.escapeName(regName);
+            string reg = "%" + regName;
+            return reg;
         } else {
+            int regName = self.unnamedRegisterCount;
             self.unnamedRegisterCount += 1;
+            return regName;
         }
-        if self.variableNames.hasKey(regName) {
-            int count = self.variableNames.get(regName);
-            self.variableNames[regName] = count + 1; // increment the count of the base name
-            string newName = regName + "." + count.toString();
-            while self.variableNames.hasKey(newName) {
-                count += 1;
-                newName = regName + "." + count.toString();
-            }
-            regName = newName;
-            self.variableNames[regName] = 1; // save the augmented name in case user use the same  
-        }
-        else {
-            self.variableNames[regName] = 1;
+    }
+
+    function escapeName(string name) returns string {
+        if regex:matches(name, "[-a-zA-Z$._][-a-zA-Z$._0-9]*") {
+            return name;
+        } else {
+            return "\"" + name + "\"";
```

Comment:
Some characters will need quoting with `\`

Reply:
Fixed. We are wrapping names with `"` if it doesn't match the given pattern (similar to what `jni.llvm` does) only character that I think we have to escape is `"` it self.